### ISSUE TITLE
Car Thing shows nothing playing

### ISSUE THREAD

**Issue by TechGeek01** [BluDood/GlanceThing#17]
**Describe the bug**
When playing music, after resuming from being paused for a while, sometimes the Car Thing does not recognize there is music playing. Restarting the server, as well as rebooting the Car Thing do not fix this issue. The fix seems to be to close and reopen Spotify.

**To Reproduce**
1. Play music
2. Pause music, and be idle for some time
3. Resume playback
4. Occasionally, when resuming playback, the Car Thing doesn't recognize the lack of idle state, and still shows idle.

**Expected behavior**
The Car Thing should detect the media playback, and behave accordingly

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11 Pro for Workstations, 23H2, build 22631.4391
 - GlanceThing version: v0.0.6

**Comment by BluDood** [comment]
Could you check the logs (`%appdata%/glancething/glancething.log`), and see if there are any errors about refreshing the Spotify token? 
On another note, I have noticed that sometimes Spotify fails to send an idle status when nothing has been playing for a while, could be fixed by sending a request every minute or so to update it

**Comment by BluDood** [comment]
That's weird. I should probably print a more detailed error when one comes in, since it's pretty hard to diagnose the issue from here. How long approximately does it take for this to start happening?

**Comment by TechGeek01** [comment]
I've had it happen a few times. Maybe once every day or 2 or so. Usually after leaving it idle for a while.

I do have the screensaver enabled, but it never does anything, despite the idle detection working (and showing the "nothing playing" bit properly. Computer itself has a screensaver, but is basically always on, never goes into sleep mode. Not sure if that's related or not.

**Comment by BluDood** [comment]
> I've had it happen a few times. Maybe once every day or 2 or so. Usually after leaving it idle for a while.
> 
> I do have the screensaver enabled, but it never does anything, despite the idle detection working (and showing the "nothing playing" bit properly. Computer itself has a screensaver, but is basically always on, never goes into sleep mode. Not sure if that's related or not.

Yeah, the screensaver is only automatically enabled when the desktop app is disconnected, can be manually enabled in the system menu. Screensaver on your computer shouldn't matter. I'll try to replicate the error myself, and at least implement some better error logs :)

**Comment by BluDood** [comment]
Closing this in favor of #19 as they're both related to connectivity issues with Spotify
